blog-server-services/src/impls/rbatis_post_service.rs

```diff
@@ -94,13 +121,55 @@ impl PostService for RbatisPostService {
         Ok(Post::count(&self.rb).await?)
     }
     async fn posts(&self, offset: &i64, limit: &i64) -> DResult<Vec<Post>> {
-        Ok(Post::select_all_with_limit_and_offset(&self.rb, limit, offset).await?)
+        let mut posts = Post::select_posts(&self.rb, limit, offset).await?;
+
+        let post_ids = posts.iter().map(|post| post.id).collect();
+        let grouped_tags: HashMap<i64, Vec<Tag>> = Post::select_tags_by_posts(&self.rb, post_ids)
+            .await?
+            .into_iter()
+            .fold(HashMap::new(), |mut map, dto| {
+                let key = dto.post_id;
+                let tag = dto.into_tag();
+                map.entry(key).or_insert_with(Vec::new).push(tag);
+                map
+            });
+
+        for post in posts.iter_mut() {
+            match grouped_tags.get(&post.id) {
+                Some(tags) => post.feed(tags),
+                None => {}
+            }
+        }
+
+        Ok(posts)
     }
     async fn post_by_id(&self, id: &i64) -> DResult<Option<Post>> {
-        Ok(Post::select_by_id(&self.rb, id).await?)
+        let mut post = Post::select_by_id(&self.rb, id)
+            .await?
+            .ok_or::<DError>("Could not find post by id".into())?;
```

Comment:
pog

Reply:
dbError replaces with option matching, now we return Ok(None) so not_found should be processed correctly
![image](https://github.com/tikitko/blog-server/assets/52403678/3adce33e-8abb-4aed-bcbc-0b648155f893)